src/main/java/gregtech/loaders/oreprocessing/DecompositionRecipeHandler.java

```diff
@@ -57,11 +57,50 @@ private static void processDecomposition(OrePrefix decomposePrefix, IMaterial<?>
             }
         }
 
+        //only reduce items
+        if (decomposePrefix != null) {
+            //calculate lowest common denominator
+            List<Integer> materialAmounts = new ArrayList<>();
+            materialAmounts.add(totalInputAmount);
+            outputs.forEach(itemStack -> materialAmounts.add(itemStack.getCount()));
+            fluidOutputs.forEach(fluidStack -> materialAmounts.add(fluidStack.amount / 1000));
+
+            int highestDivisor = 1;
+            for (int i = 2; i <= 64; i++) {
+                if (isEveryMaterialReducible(i, materialAmounts))
```

Comment:
This is a bit of an optimization thing in my mind, so let me just brainstorm here real quick and see what you think:

On average I would say this highest divisor would be less than 10 for most materials. Since that is the case, we can halve the algorithmically efficiency here by running an additional method to break if there are no material components greater than the divisor. If the average were 32+, then we wouldn't want to do this since it is adding twice the work to the list.

Lets say
`isEveryMaterialReducable` takes N time. So this loop is roughly 64N. If we add a method in the loop like `isEveryMaterialLessThan`, also N time, then at worst this is 128N. But, since the rough average (in my opinion) for material component sizes is 10, we can reduce this from 64N -> 20N by cutting out many iterations with this additional method.